app/Main.hs

```diff
@@ -1,6 +1,85 @@
 module Main where
 
-import Text.Printf (printf)
+import Entities
+import Control.Monad.IO.Class  (liftIO)
+import Database.Persist
+import Database.Persist.Sqlite
+import Text.Megaparsec
+import FooParser (parseSections, Section(..))
+import qualified Database.Esqueleto as E
 
 main :: IO ()
-main = printf "foo"
+main = do
+  fileContents <- readFile "specs.foo"
+
+  -- parse specs.foo
+  let sections = case runParser parseSections "specs.foo" fileContents of
+        Left err        -> error $ "could not parse .foo file with error: " ++ show err
```

Comment:
This kind of thing is reasonable to do at the top level in an example program, but it would be a good place to talk briefly about avoiding partial functions.